File: Kattis/sim/sim.cpp
#include <iostream>

using namespace std;
int main()
{
	int n;
	cin >> n;
	string throwaway;
	getline(cin,throwaway);
	for(int i = 0;i<n;i++)
	{
		string line;
		getline(cin,line);
		int chars = line.length();
		string text = "";
		int index = 0;
		for(int j = 0;j<chars;j++)
		{
			char character = line[j];
			if(character == '<')
			{
				if(index>0)
				{
					text = text.substr(0,index-1) + text.substr(index,text.length());
					index--;
				}
			}
			else if(character == '[')
			{
				index = 0;
			}
			else if(character == ']')
			{
				index = text.length();
			}
			else
			{
				if(text.length()==0) 
				{
					text = text+character;
					index++;
				}
				else
				{
					text = text.substr(0,index) + character + text.substr(index,text.length());
					index++;
				}
			}
		}
		cout<<text<<endl;
	}
	return 0;
}

File: Kattis/sim/sim.py
import sys
n = int(sys.stdin.readline())
for i in range(0,n):
	line = sys.stdin.readline().strip()
	text = ''
	index = 0
	for char in line:
		if(char == '<'):
			if(not index==0):
				text = text[:index-1] + text[index:]
				index-=1
		elif(char == '['):
			index = 0
		elif(char == ']'):
			index = len(text)
		else:
			text = text[:index] + char + text[index:]
			index+=1
	print(text)
				
